de

```diff
@@ -19,18 +19,17 @@ function fail {
     exit 1
 }
 function store_variable {
-    echo "$1"
     echo -n "${2}=" >> ~/.de
-    if ${3}; then read -s ${2}; else read ${2}; fi
+    if ${3}; then read -sp "${1}" ${2}; echo ""; else read -p "${1}" ${2}; fi
     echo "${!2}" >> ~/.de
 }
 
 if [ ! -f ~/.de ]; then
     echo "No config file found. Creating one for you"
     touch ~/.de
-    store_variable "Input the passcode of your origin network." "start_network_psk" true
-    store_variable "Input the name of your robot's network." "robot_network" false
-    store_variable "Input the passcode of your robot's network." "robot_network_psk" true
+    store_variable "Input the passcode of your origin network. " "start_network_psk" true
+    store_variable "Input the name of your robot's network. " "robot_network" false
+    store_variable "Input the passcode of your robot's network. " "robot_network_psk" true
```

Comment:
I would recommend adding in the space (and possibly the period as well" in the store_variable function since you're adding it every time.